### PR TITLE
Add new RRSwISC6to18E3r1 ocean and sea-ice mesh

### PR DESCRIPTION
**Pull request by xylar**
Long name: RRSwISC6to18L80E3SMv3r1

This RRS (Rossby-radius scaled) mesh has:
* 6 km resolution near the poles
* 18 km resolution at the equator

This is a proposed E3SM v3 (E3) high resolution (near-eddy-resolving) mesh.  This is revision 1 (r1) of the mesh, since we caught and fixed the issues with Greenland topography (seen in earlier revisions of EC and Icos meshes) before any substantial testing was performed.

The mesh was created using [compass](https://github.com/MPAS-Dev/compass), specifically this PR: 
https://github.com/MPAS-Dev/compass/pull/668
A compass tag will be created for the mesh as soon as the PR is merged.

The mesh and forthcoming simulation results will be reviewed here:
https://acme-climate.atlassian.net/wiki/spaces/OO/pages/3894149135/Review+RRSwISC6to18E3r1

G- and B-case simulations will begin shortly and analysis will be posted here and on the review page as soon as it is available.

### CODE REVIEW

**Comment by xylar** [comment]
Moved from https://github.com/E3SM-Ocean-Discussion/E3SM/pull/59, which has a small amount of relevant discussion.

File: cime_config/config_grids.xml [hunk 1]
```diff
@@ -2771,6 +2806,13 @@
       <desc>ECwISC30to60E3r2 is a MPAS ocean grid generated with the jigsaw/compass process using the eddy closure density function that has 30 km gridcells at the equator, 60 km at mid-latitudes, and 35 km at high latitudes. Additionally, it has ocean in ice-shelf cavities:</desc>
     </domain>
 
+    <domain name="RRSwISC6to18E3r1">
+      <nx>4085395</nx>
+      <ny>1</ny>
+      <file grid="ice|ocn">$DIN_LOC_ROOT/share/domains/domain.ocn.RRSwISC6to18E3r1.230914.nc</file>
+      <desc>RRSwISC6to18E3r1 is a MPAS ocean grid generated with the jigsaw/compass process using XXXXX. Additionally, it has ocean in ice-shelf cavities:</desc>
```
Comment (rljacob) [inline cime_config/config_grids.xml:2813]: Should something replace "XXXXX" ?

File: cime_config/config_grids.xml [hunk 2]
```diff
@@ -3265,6 +3307,14 @@
       <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/ECwISC30to60E3r2/map_ECwISC30to60E3r2_to_ne30pg2_traave.20231018.nc</map>
     </gridmap>
 
+    <gridmap atm_grid="ne30np4.pg2" ocn_grid="RRSwISC6to18E3r1">
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/ne30pg2/map_ne30pg2_to_RRSwISC6to18E3r1_mono.230914.nc</map>
```
Comment (rljacob) [inline cime_config/config_grids.xml:3311]: This should be renamed to "traave" instead of "mono".

File: cime_config/config_grids.xml (hunk 1 above)
Comment (xylar) [reply cime_config/config_grids.xml:2813]: Thanks @rljacob.  We'll fix that.

**Review by rljacob — CHANGES_REQUESTED** [review]
Lots of map names need to change to follow our new convenction.  Anything with "pg2" and "mono" is really "traave".  The "bilin" should be "trbilin".

File: cime_config/config_grids.xml (hunk 2 above)
Comment (jonbob) [reply cime_config/config_grids.xml:3311]: @rljacob -- I'll regenerate the mapping files to match the new algorithms and naming convention

**Comment by mt5555** [comment]
the atm2ocn flux map might have the blocky grid imprinting issue in coupled simulations (for the low-res ne30pg2 atmosphere coupled to this new ocean grid)   
(i.e. https://acme-climate.atlassian.net/wiki/spaces/ECG/pages/2350776353/2-17-2021+Cryosphere+Simulation+Campaign ) 

if this is used for coupled simulations we may want to consider upgrading the atm2ocn map to a "nonlinear" flux map:
https://acme-climate.atlassian.net/wiki/spaces/COM/pages/3771301899/Coupler+High-order+property-preserving+maps

**Comment by xylar** [comment]
@jonbob, I think these comments are things you may be able to best address.  Feel free to push to this branch.

**Review by proteanplanet — APPROVED** [review]
I have reviewed bathymetry, narrow passages, and ice shelf cavities both numerically and visually.  Summary plots are provided below for packaging with this PR and later referral.  Approved for the magnificent quality of the mesh. 

image

image

image

image

image

image

image

image

MPAS_OceanIce_ECwISC30to60E3r2_RRSwISC6to18E3r1_zoom_bathymetry_31

MPAS_OceanIce_ECwISC30to60E3r2_RRSwISC6to18E3r1_zoom_bathymetry_32

MPAS_OceanIce_ECwISC30to60E3r2_RRSwISC6to18E3r1_zoom_bathymetry_33

MPAS_OceanIce_ECwISC30to60E3r2_RRSwISC6to18E3r1_zoom_bathymetry_34

MPAS_OceanIce_ECwISC30to60E3r2_RRSwISC6to18E3r1_zoom_bathymetry_35

MPAS_OceanIce_ECwISC30to60E3r2_RRSwISC6to18E3r1_zoom_bathymetry_36

MPAS_OceanIce_ECwISC30to60E3r2_RRSwISC6to18E3r1_zoom_bathymetry_45

MPAS_OceanIce_ECwISC30to60E3r2_RRSwISC6to18E3r1_zoom_bathymetry_46

MPAS_OceanIce_ECwISC30to60E3r2_RRSwISC6to18E3r1_zoom_bathymetry_47

**Comment by xylar** [comment]
@proteanplanet, I appreciate you posting these plots!  I'd request that you hold off on approving the mesh, though, until we've done some forward simulations (preferably fully coupled) so we can see what the behavior is like.  I have put this up so that reviewers can approve this PR and the associated columns in the confluence document together.  I didn't have a chance to explain this to you all, I'm afraid.  I was just asked to make this mesh the next priority earlier today.

**Comment by xylar** [comment]
I'm putting this in draft mode because I just discovered a bug in `landIceMask` (which affects the initial condition as well as several mapping files) over the weekend.

**Comment by xylar** [comment]
Closing this because of the `landIceMask` bug.  We will generate a new version with the bug fixed in the new year.

**Pull request closed without merging**